### PR TITLE
Introduce pre-commit hooks

### PR DESCRIPTION
**Pull request by mulhod**
Hooks are installed as a development dependency, so this shouldn't affect anything in terms of packaging.

I have left out `black` for now because I don't want this getting too big and controversial. I will do that next separately so that we can all see what exactly will happen.

### CODE REVIEW

**Comment by mulhod** [comment]
Addresses #646

**Review by desilinguist — COMMENTED** [review]
Should we exclude the CSV/TSV/JSON files from the test data from the auto-formatting fixes?

**Review by desilinguist — CHANGES_REQUESTED** [review]
Everything else looks good to me! This will be quite useful! Just that one minor suggestion.

**Comment by mulhod** [comment]
> Should we exclude the CSV/TSV/JSON files from the test data from the auto-formatting fixes?

I have gone ahead and excluded these files from the EOF file hook.

**Review by desilinguist — APPROVED** [review]
LGTM!

File: skll/learner/__init__.py [hunk 1]
```diff
@@ -11,78 +11,83 @@
...
-                              RandomForestRegressor)
-from sklearn.feature_extraction import FeatureHasher
+from sklearn.dummy import DummyClassifier, DummyRegressor  # noqa: F401
+from sklearn.ensemble import (
+    AdaBoostClassifier,
+    AdaBoostRegressor,
+    GradientBoostingClassifier,
+    GradientBoostingRegressor,
+    RandomForestClassifier,
+    RandomForestRegressor,
```
Comment (bndgyawali) [inline skll/learner/__init__.py:29]: do we need a comma at the end?

File: skll/learner/__init__.py [hunk 2]
```diff
@@ -11,78 +11,83 @@
...
+    Lars,
+    Lasso,
+    LinearRegression,
+    LogisticRegression,
+    RANSACRegressor,
+    Ridge,
+    RidgeClassifier,
+    SGDClassifier,
+    SGDRegressor,
+    TheilSenRegressor,
```
Comment (bndgyawali) [inline skll/learner/__init__.py:47]: here too.

File: skll/learner/__init__.py [hunk 3]
```diff
@@ -11,78 +11,83 @@
...
+    compute_num_folds_from_example_counts,
+    get_acceptable_classification_metrics,
+    get_acceptable_regression_metrics,
+    get_predictions,
+    load_custom_learner,
+    rescaled,
+    setup_cv_fold_iterator,
+    setup_cv_split_iterator,
+    train_and_score,
+    write_predictions,
```
Comment (bndgyawali) [inline skll/learner/__init__.py:89]: and here

File: skll/learner/utils.py [hunk 4]
```diff
@@ -22,21 +21,27 @@
...
-                             confusion_matrix,
-                             precision_recall_fscore_support)
-from sklearn.model_selection import (KFold,
-                                     LeaveOneGroupOut,
-                                     ShuffleSplit,
-                                     StratifiedKFold)
+from sklearn.metrics import (
+    accuracy_score,
+    confusion_matrix,
+    precision_recall_fscore_support,
```
Comment (bndgyawali) [inline skll/learner/utils.py:27]: seems it is having comma everywhere in this type of import statements.

**Review by bndgyawali — APPROVED** [review]
I have a comment, but seems like that is how black does. 
Else everything looks good to me.

**Review by aoifecahill — APPROVED** [review]

**Pull request merged**